ui/src/containers/AppDetailsContainer.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import AppsTable from '../components/AppsTable';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { Checkbox } from '@patternfly/react-core';
+import { getApps, reverseAppDetailsTableSort } from '../actions/actions-ui';
+
+export class AppDetailsContainer extends React.Component {
+  constructor (props) {
+    super(props);
+    this.onSort = this.onSort.bind(this);
+    this.onRowClick = this.onRowClick.bind(this);
+    this.getTable = this.getTable.bind(this);
+  }
+
+  componentWillMount () {
+    //this.props.getApps();
+  }
+
+  handleChange = (e) => {};
+
+  onRowClick = (event, rowId, props) => {
+    // const app = this.props.apps.data.filter((app) => {
+    //   return app.appName === rowId[0];
+    // });
+    // const id = app[0].id;
+    // const path = '/app/' + id;
+    // this.props.history.push(path);
+  };
+
+  onSort = (_event, index) => {
+    this.props.reverseAppDetailsTableSort(index);
+  };
+
+  getTable = () => {
+    const renderedRows = [];
+    for (let i = 0; i < this.props.apps.length; i++) {
+      const tempRow = [];
+      tempRow[0] = this.props.apps[i][0];
+      tempRow[1] = this.props.apps[i][1];
+      tempRow[2] = this.props.apps[i][2];
+      tempRow[3] = this.props.apps[i][3];
+      const cb = (
+        <Checkbox
+          label=""
+          isChecked={this.props.apps[i][4]}
+          onChange={this.handleChange}
+          aria-label="controlled checkbox example"
+          id={i}
+        />
+      );
+      tempRow[4] = cb;
+      tempRow[5] = this.props.apps[i][5];
+      renderedRows.push(tempRow);
+    }
+
+    return (
+      <div className="apps-table">
+        <AppsTable
+          caption="Deployed Versions"
+          columns={this.props.columns}
+          rows={renderedRows}
+          sortBy={this.props.sortBy}
+          onSort={this.onSort}
+          onRowClick={this.onRowClick}
+        />
+      </div>
+    );
+  };
+
+  render () {
+    if (this.props.isAppsRequestFailed) {
```

Comment:
I believe we can reuse this. If the GET app doesn't work we could use `isAppRequestFailed` as apposed to `isAppsRequestFailed`

Reply:
agree - we can include that in the PR that adds the API get call

Reply:
resolving as should be addressed in AEROGEAR-8466

Reply:
This should probably redirect to a 404 Page if the app doesn't exist.

Reply:
@craicoverflow I left a comment in one of the trello cards about the blank slate/not found experience. Going to ask about that in the review as we haven't gotten a reply as yet.